[Bug] Json serialization issue when using Redis as storage.

### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/eventmesh/issues?q=is%3Aissue) and found no similar issues.


### Environment

Mac

### EventMesh version

master

### What happened

when I start the TCP demo and set Redis as storage, the runtime throw json parse exception because the body have not been serialized correctly.
![image](https://user-images.githubusercontent.com/11908658/226967562-4ccef164-1de9-45a7-a239-c71478e647ac.png)

see data in Redis
![image](https://user-images.githubusercontent.com/11908658/226967461-2b2decb3-c633-4934-8487-d8d3048fd746.png)


### How to reproduce

1. set `eventMesh.storage.plugin.type=redis` in runtime config.
2. add your redis config in `redis-client.properties.
3. start `runtime` via `org.apache.eventmesh.starter.StartUp`.
4. start subscribe demo via `org.apache.eventmesh.tcp.demo.sub.eventmeshmessage.AsyncSubscribe`.
5. start tcp publish demo via `org.apache.eventmesh.tcp.demo.pub.eventmeshmessage.AsyncPublish`.
6. now the error will occur.

### Debug logs
```
Caused by: com.fasterxml.jackson.core.JsonParseException: Unrecognized token 'testAsyncMessage': was expecting (JSON String, Number, Array, Object or token 'null', 'true' or 'false')
 at [Source: (byte[])"{"specversion":"1.0","id":"5683320505","source":"/","type":"eventmeshmessage","subject":"TEST-TOPIC-TCP-ASYNC","protocolversion":"1.0","reqeventmesh2mqtimestamp":"1679593289588","propertymessagereplyto":"notnull","reqsendeventmeship":"[fe80:0:0:0:aede:48ff:fe00:1122%en5]","replyto":"localhost@ProducerGroup-producerPool-9-access#V1_4_0#CI","protocoldesc":"tcp","protocoltype":"eventmeshmessage","ttl":"30000","reqc2eventmeshtimestamp":"1679593289587","data":testAsyncMessage}"; line: 1, column: 477]
Caused by: com.fasterxml.jackson.core.JsonParseException: Unrecognized token 'testAsyncMessage': was expecting (JSON String, Number, Array, Object or token 'null', 'true' or 'false')

	at com.fasterxml.jackson.core.JsonParser._constructError(JsonParser.java:2391) ~[jackson-core-2.13.0.jar:2.13.0]
	at com.fasterxml.jackson.core.base.ParserMinimalBase._reportError(ParserMinimalBase.java:745) ~[jackson-core-2.13.0.jar:2.13.0]
	at com.fasterxml.jackson.core.json.UTF8StreamJsonParser._reportInvalidToken(UTF8StreamJsonParser.java:3634) ~[jackson-core-2.13.0.jar:2.13.0]
	at com.fasterxml.jackson.core.json.UTF8StreamJsonParser._reportInvalidToken(UTF8StreamJsonParser.java:3607) ~[jackson-core-2.13.0.jar:2.13.0]
	at com.fasterxml.jackson.core.json.UTF8StreamJsonParser._matchToken2(UTF8StreamJsonParser.java:2948) ~[jackson-core-2.13.0.jar:2.13.0]
	at com.fasterxml.jackson.core.json.UTF8StreamJsonParser._matchTrue(UTF8StreamJsonParser.java:2882) ~[jackson-core-2.13.0.jar:2.13.0]
	at com.fasterxml.jackson.core.json.UTF8StreamJsonParser.nextFieldName(UTF8StreamJsonParser.java:1115) ~[jackson-core-2.13.0.jar:2.13.0]
	at com.fasterxml.jackson.databind.deser.std.BaseNodeDeserializer._deserializeContainerNoRecursion(JsonNodeDeserializer.java:440) ~[jackson-databind-2.13.0.jar:2.13.0]
	at com.fasterxml.jackson.databind.deser.std.JsonNodeDeserializer$ObjectDeserializer.deserialize(JsonNodeDeserializer.java:122) ~[jackson-databind-2.13.0.jar:2.13.0]
	at com.fasterxml.jackson.databind.deser.std.JsonNodeDeserializer$ObjectDeserializer.deserialize(JsonNodeDeserializer.java:105) ~[jackson-databind-2.13.0.jar:2.13.0]
	at com.fasterxml.jackson.databind.DeserializationContext.readValue(DeserializationContext.java:966) ~[jackson-databind-2.13.0.jar:2.13.0]
	at com.fasterxml.jackson.databind.DeserializationContext.readValue(DeserializationContext.java:953) ~[jackson-databind-2.13.0.jar:2.13.0]
	at io.cloudevents.jackson.CloudEventDeserializer.deserialize(CloudEventDeserializer.java:201) ~[cloudevents-json-jackson-2.2.0.jar:?]
	at io.cloudevents.jackson.CloudEventDeserializer.deserialize(CloudEventDeserializer.java:40) ~[cloudevents-json-jackson-2.2.0.jar:?]
	at com.fasterxml.jackson.databind.deser.DefaultDeserializationContext.readRootValue(DefaultDeserializationContext.java:322) ~[jackson-databind-2.13.0.jar:2.13.0]
	at com.fasterxml.jackson.databind.ObjectMapper._readMapAndClose(ObjectMapper.java:4675) ~[jackson-databind-2.13.0.jar:2.13.0]
	at com.fasterxml.jackson.databind.ObjectMapper.readValue(ObjectMapper.java:3691) ~[jackson-databind-2.13.0.jar:2.13.0]
	at io.cloudevents.jackson.JsonFormat.deserialize(JsonFormat.java:99) ~[cloudevents-json-jackson-2.2.0.jar:?]
	at org.apache.eventmesh.storage.redis.cloudevent.CloudEventCodec.lambda$static$1(CloudEventCodec.java:48) ~[eventmesh-storage-redis-1.8.0-release.jar:1.8.0-release]
	at org.redisson.client.handler.CommandDecoder.decode(CommandDecoder.java:383) ~[redisson-3.17.3.jar:3.17.3]
	at org.redisson.client.handler.CommandDecoder.decodeList(CommandDecoder.java:427) ~[redisson-3.17.3.jar:3.17.3]
	at org.redisson.client.handler.CommandDecoder.decode(CommandDecoder.java:392) ~[redisson-3.17.3.jar:3.17.3]
	at org.redisson.client.handler.CommandPubSubDecoder.decodeCommand(CommandPubSubDecoder.java:86) ~[redisson-3.17.3.jar:3.17.3]
	at org.redisson.client.handler.CommandDecoder.decode(CommandDecoder.java:137) ~[redisson-3.17.3.jar:3.17.3]
	at org.redisson.client.handler.CommandDecoder.decode(CommandDecoder.java:94) ~[redisson-3.17.3.jar:3.17.3]
	at io.netty.handler.codec.ByteToMessageDecoder.decodeRemovalReentryProtection(ByteToMessageDecoder.java:510) ~[netty-codec-4.1.79.Final.jar:4.1.79.Final]
	at io.netty.handler.codec.ReplayingDecoder.callDecode(ReplayingDecoder.java:366) ~[netty-codec-4.1.79.Final.jar:4.1.79.Final]
	... 17 more
```

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

--- a/eventmesh-common/src/main/java/org/apache/eventmesh/common/Constants.java
+++ b/eventmesh-common/src/main/java/org/apache/eventmesh/common/Constants.java
@@ -28,6 +28,8 @@ public class Constants {
 
     public static final String DATE_FORMAT_DEFAULT = "yyyy-MM-dd HH:mm:ss";
 
+    public static final String DATA_CONTENT_TYPE = "DataContentType";
+
     public static final String LANGUAGE_JAVA = "JAVA";
 
     public static final String HTTP_PROTOCOL_PREFIX = "http://";

--- a/eventmesh-common/src/main/java/org/apache/eventmesh/common/Constants.java
+++ b/eventmesh-common/src/main/java/org/apache/eventmesh/common/Constants.java
@@ -28,7 +28,7 @@ public class Constants {
 
     public static final String DATE_FORMAT_DEFAULT = "yyyy-MM-dd HH:mm:ss";
 
-    public static final String DATA_CONTENT_TYPE = "DataContentType";
+    public static final String DATA_CONTENT_TYPE = "datacontenttype";
 
     public static final String LANGUAGE_JAVA = "JAVA";
 

--- a/eventmesh-common/src/main/java/org/apache/eventmesh/common/protocol/tcp/EventMeshMessage.java
+++ b/eventmesh-common/src/main/java/org/apache/eventmesh/common/protocol/tcp/EventMeshMessage.java
@@ -31,5 +31,6 @@ public class EventMeshMessage {
 
     private String topic;
     private Map<String, String> properties = new ConcurrentHashMap<>();
+    private Map<String, String> headers = new ConcurrentHashMap<>();
     private String body;
 }

--- a/eventmesh-examples/src/main/java/org/apache/eventmesh/tcp/common/EventMeshTestUtils.java
+++ b/eventmesh-examples/src/main/java/org/apache/eventmesh/tcp/common/EventMeshTestUtils.java
@@ -21,6 +21,7 @@ import static org.apache.eventmesh.common.protocol.tcp.Command.RESPONSE_TO_SERVE
 
 import org.apache.eventmesh.client.tcp.common.EventMeshCommon;
 import org.apache.eventmesh.client.tcp.common.MessageUtils;
+import org.apache.eventmesh.common.Constants;
 import org.apache.eventmesh.common.ExampleConstants;
 import org.apache.eventmesh.common.protocol.tcp.Command;
 import org.apache.eventmesh.common.protocol.tcp.EventMeshMessage;
@@ -125,6 +126,7 @@ public class EventMeshTestUtils {
         mqMsg.getProperties().put(UtilsConstants.MSG_TYPE, "persistent");
         mqMsg.getProperties().put(UtilsConstants.TTL, DEFAULT_TTL_MS);
         mqMsg.getProperties().put(UtilsConstants.KEYS, generateRandomString(16));
+        mqMsg.getHeaders().put(Constants.DATA_CONTENT_TYPE, "text/plain");
         mqMsg.setBody("testSyncRR");
         return mqMsg;
     }
@@ -136,6 +138,7 @@ public class EventMeshTestUtils {
         mqMsg.getProperties().put(UtilsConstants.REPLY_TO, "localhost@ProducerGroup-producerPool-9-access#V1_4_0#CI");
         mqMsg.getProperties().put(UtilsConstants.TTL, DEFAULT_TTL_MS);
         mqMsg.getProperties().put(UtilsConstants.PROPERTY_MESSAGE_REPLY_TO, "notnull");
+        mqMsg.getHeaders().put(Constants.DATA_CONTENT_TYPE, "text/plain");
         mqMsg.setBody("testAsyncRR");
         return mqMsg;
     }
@@ -146,6 +149,7 @@ public class EventMeshTestUtils {
         mqMsg.getProperties().put(UtilsConstants.REPLY_TO, "localhost@ProducerGroup-producerPool-9-access#V1_4_0#CI");
         mqMsg.getProperties().put(UtilsConstants.TTL, DEFAULT_TTL_MS);
         mqMsg.getProperties().put(UtilsConstants.PROPERTY_MESSAGE_REPLY_TO, "notnull");
+        mqMsg.getHeaders().put(Constants.DATA_CONTENT_TYPE, "text/plain");
         mqMsg.setBody(ASYNC_MSG_BODY);
         return mqMsg;
     }
@@ -156,6 +160,7 @@ public class EventMeshTestUtils {
         mqMsg.getProperties().put(UtilsConstants.REPLY_TO, "localhost@ProducerGroup-producerPool-9-access#V1_4_0#CI");
         mqMsg.getProperties().put(UtilsConstants.TTL, DEFAULT_TTL_MS);
         mqMsg.getProperties().put(UtilsConstants.PROPERTY_MESSAGE_REPLY_TO, "notnull");
+        mqMsg.getHeaders().put(Constants.DATA_CONTENT_TYPE, "text/plain");
         mqMsg.setBody(ASYNC_MSG_BODY);
         return mqMsg;
     }

--- a/eventmesh-protocol-plugin/eventmesh-protocol-meshmessage/src/main/java/org/apache/eventmesh/protocol/meshmessage/resolver/tcp/TcpMessageProtocolResolver.java
+++ b/eventmesh-protocol-plugin/eventmesh-protocol-meshmessage/src/main/java/org/apache/eventmesh/protocol/meshmessage/resolver/tcp/TcpMessageProtocolResolver.java
@@ -35,6 +35,7 @@ import java.util.Objects;
 import io.cloudevents.CloudEvent;
 import io.cloudevents.SpecVersion;
 import io.cloudevents.core.builder.CloudEventBuilder;
+import org.assertj.core.util.Strings;
 
 public class TcpMessageProtocolResolver {
 
@@ -114,6 +115,10 @@ public class TcpMessageProtocolResolver {
 
         Package pkg = new Package();
         pkg.setBody(eventMeshMessage);
+
+        if (!Strings.isNullOrEmpty(cloudEvent.getDataContentType())) {
+            eventMeshMessage.getHeaders().put(Constants.DATA_CONTENT_TYPE, cloudEvent.getDataContentType());
+        }
 
         return pkg;
     }

--- a/eventmesh-protocol-plugin/eventmesh-protocol-meshmessage/src/main/java/org/apache/eventmesh/protocol/meshmessage/resolver/tcp/TcpMessageProtocolResolver.java
+++ b/eventmesh-protocol-plugin/eventmesh-protocol-meshmessage/src/main/java/org/apache/eventmesh/protocol/meshmessage/resolver/tcp/TcpMessageProtocolResolver.java
@@ -35,7 +35,6 @@ import java.util.Objects;
 import io.cloudevents.CloudEvent;
 import io.cloudevents.SpecVersion;
 import io.cloudevents.core.builder.CloudEventBuilder;
-import org.assertj.core.util.Strings;
 
 public class TcpMessageProtocolResolver {
 
@@ -116,7 +115,7 @@ public class TcpMessageProtocolResolver {
         Package pkg = new Package();
         pkg.setBody(eventMeshMessage);
 
-        if (!Strings.isNullOrEmpty(cloudEvent.getDataContentType())) {
+        if (!StringUtils.isBlank(cloudEvent.getDataContentType())) {
             eventMeshMessage.getHeaders().put(Constants.DATA_CONTENT_TYPE, cloudEvent.getDataContentType());
         }
 

--- a/eventmesh-protocol-plugin/eventmesh-protocol-meshmessage/src/main/java/org/apache/eventmesh/protocol/meshmessage/resolver/tcp/TcpMessageProtocolResolver.java
+++ b/eventmesh-protocol-plugin/eventmesh-protocol-meshmessage/src/main/java/org/apache/eventmesh/protocol/meshmessage/resolver/tcp/TcpMessageProtocolResolver.java
@@ -38,7 +38,6 @@ import io.cloudevents.core.builder.CloudEventBuilder;
 
 public class TcpMessageProtocolResolver {
 
-
     public static CloudEvent buildEvent(Header header, EventMeshMessage message) throws ProtocolHandleException {
 
         CloudEventBuilder cloudEventBuilder;
@@ -67,7 +66,6 @@ public class TcpMessageProtocolResolver {
 
         } else if (StringUtils.equals(SpecVersion.V03.toString(), protocolVersion)) {
             cloudEventBuilder = CloudEventBuilder.v03();
-
         } else {
             throw new ProtocolHandleException(String.format("Unsupported protocolVersion: %s", protocolVersion));
         }
@@ -76,9 +74,12 @@ public class TcpMessageProtocolResolver {
             .withId(header.getSeq())
             .withSource(URI.create("/"))
             .withType("eventmeshmessage")
-            .withDataContentType("text/plain")
             .withSubject(topic)
-            .withData(content.getBytes(StandardCharsets.UTF_8));
+            .withData(content.getBytes(Constants.DEFAULT_CHARSET));
+
+        if (message.getHeaders().containsKey(Constants.DATA_CONTENT_TYPE)) {
+            cloudEventBuilder.withDataContentType(message.getHeaders().get(Constants.DATA_CONTENT_TYPE));
+        }
 
         for (String propKey : header.getProperties().keySet()) {
             try {

--- a/eventmesh-protocol-plugin/eventmesh-protocol-meshmessage/src/main/java/org/apache/eventmesh/protocol/meshmessage/resolver/tcp/TcpMessageProtocolResolver.java
+++ b/eventmesh-protocol-plugin/eventmesh-protocol-meshmessage/src/main/java/org/apache/eventmesh/protocol/meshmessage/resolver/tcp/TcpMessageProtocolResolver.java
@@ -39,7 +39,6 @@ import io.cloudevents.core.builder.CloudEventBuilder;
 public class TcpMessageProtocolResolver {
 
     public static CloudEvent buildEvent(Header header, EventMeshMessage message) throws ProtocolHandleException {
-
         CloudEventBuilder cloudEventBuilder;
 
         String protocolType = header.getProperty(Constants.PROTOCOL_TYPE).toString();

--- a/eventmesh-protocol-plugin/eventmesh-protocol-meshmessage/src/main/java/org/apache/eventmesh/protocol/meshmessage/resolver/tcp/TcpMessageProtocolResolver.java
+++ b/eventmesh-protocol-plugin/eventmesh-protocol-meshmessage/src/main/java/org/apache/eventmesh/protocol/meshmessage/resolver/tcp/TcpMessageProtocolResolver.java
@@ -76,6 +76,7 @@ public class TcpMessageProtocolResolver {
             .withId(header.getSeq())
             .withSource(URI.create("/"))
             .withType("eventmeshmessage")
+            .withDataContentType("text/plain")
             .withSubject(topic)
             .withData(content.getBytes(StandardCharsets.UTF_8));
 

